### DISCUSSION TITLE
EEPROM LIB not working .

### DISCUSSION THREAD

**Discussion by PTT31** [stm32duino/Arduino_Core_STM32#2113] [Libraries]
#include <EEPROM.h>
#define FLASH_END  0x0800FC00UL
struct MyObject {
  float field1;
  byte field2;
  char name[10];
};

void setup() {

  Serial.begin(9600);
  while (!Serial) {
    ; // wait for serial port to connect. Needed for native USB port only
  }

  float f = 123.456f;  //Variable to store in EEPROM.
  int eeAddress = 0;   //Location we want the data to be put.

  //One simple call, with the address first and the object second.
  EEPROM.put(eeAddress, f);

  Serial.println("Written float data type!");

  /** Put is designed for use with custom structures also. **/

  //Data to store.
  MyObject customVar = {
    3.14f,
    65,
    "Working!"
  };

  eeAddress += sizeof(float); //Move address to the next byte after float 'f'.

  EEPROM.put(eeAddress, customVar);
  Serial.print("Written custom data type! \n\nView the example sketch eeprom_get to see how you can retrieve the values!");
}

void loop() {   /* Empty loop */ }

I have a fake Blue Pill from china and i can't write in flash with eeprom lib. 
thanks you.

**Comment by fpistm** [comment]
First, define added on top of the sketch is not correct, you have to redefine it using build_opt.h for example.
Finally:
> I have a fake Blue Pill from china and i can't write in flash with eeprom lib.

We do not support fake BP.
